workflow-controller/controller-service/src/main/java/io/hyscale/controller/service/ReplicaProcessingService.java

```diff
@@ -51,6 +62,19 @@
             return deployer.getReplicas(authConfig, appName, serviceName, namespace, true);
         }
     }
+    
+    public boolean podParentExists(String appName, String serviceName, String namespace) throws HyscaleException {
+        return podParentExists(null, appName, serviceName, namespace);
+    }
+    
+    public boolean podParentExists(AuthConfig authConfig, String appName, String serviceName, String namespace) throws HyscaleException {
+        authConfig = authConfig == null ? configBuilder.getAuthConfig() : authConfig;
```

Comment:
Why do we need 2 methods ? 
podParentExists => hasService() ? 
AuthConfig is anyhow necessary to fetch replicas